### ISSUE TITLE
Using Has.Count in `Assert.Multiple` re-raises nullable reference warning

### ISSUE THREAD

**Issue by smkanadl** [nunit/nunit.analyzers#436]
I recently apply a quickfix to use `Has.Count.EqualTo(1)` instead of the classical `Count, Is.EqualTo(1)`.

What I noticed afterwards is that some (to me suspicious) CS8604 are raised for the collection variable that also has an explicit `It.Not.Null` check. It only happens if the `Has.Count` is used in an `Assert.Multiple` block.

I crafted an example where it failes and two examples which compile file without any warning.

```
[Test]
public void HasCount_CreatesSuspiciousNullRefWarning()
{
    List<int>? maybeNull = null;
    Assert.That(maybeNull, Is.Not.Null);
    Assert.Multiple(() =>
    {
        Assert.That(maybeNull, Has.Count.EqualTo(2));
        Assert.That(maybeNull.Select(s => s), Is.All.EqualTo(1)); // CS8604 here, but should not!
    });
}

[Test]
public void FineWithoutHasCount()
{
    List<int>? maybeNull = null;
    Assert.That(maybeNull, Is.Not.Null);
    Assert.Multiple(() =>
    {
        Assert.That(maybeNull.Select(s => s), Is.All.EqualTo(1)); // Ok
    });
}

[Test]
public void FineWithHasCountOutsideOfAssertMultiple()
{
    List<int>? maybeNull = null;
    Assert.That(maybeNull, Is.Not.Null);
    Assert.That(maybeNull, Has.Count.EqualTo(2));
    Assert.Multiple(() =>
    {
        Assert.That(maybeNull.Select(s => s), Is.All.EqualTo(1)); // Ok
    });
}
```

Building for net5 using .net6 sdk, TestSdk 17.1, NUnit 3.13.3 and Analyzers 3.3.0.

**Comment by manfred-brands** [comment]
Thanks @smkanadl I can confirm the issue.
Weird is that when swapping the count and select, there is no problem:
```csharp
[Test]
public void FineWithHasCountLast()
{
    List<int>? maybeNull = null;
    Assert.That(maybeNull, Is.Not.Null);
    Assert.Multiple(() =>
    {
        Assert.That(maybeNull.Select(s => s), Is.All.EqualTo(1));
        Assert.That(maybeNull, Has.Count.EqualTo(2));
    });
}
```

Now to find the cause and fix. @mikkelbu You can assign this to me (I still cannot do that myself)

**Comment by smkanadl** [comment]
Thanks for taking a look.

Yes, that swap look very weird. I haven't tried it because it made no sense to me 😀.

**Comment by mikkelbu** [comment]
Thanks for taking a look at this @manfred-brands. I'll see if I can give you some permissions tonight.
